src/Microsoft.Health.Dicom.Api/Features/Converter/CustomJsonStringEnumConverterImp.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Features.Converter
+{
+    internal class CustomJsonStringEnumConverterImp<T> : JsonConverter<T>
+        where T : Enum
+    {
+        public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            EnsureArg.IsNotNull(typeToConvert, nameof(typeToConvert));
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                string content = reader.GetString();
+                if (Enum.TryParse(typeToConvert, content, true, out object result))
```

Comment:
Unfortunately, `TryParse` accepts both names and literals, right? We should have a test for that, but I'm afraid you'll need to probably cache a list of all of the names in this type and check against it.

Reply:
I'm thinking just using the default behavior which accept both name and integer string is a good or as least acceptable idea.